components/LeaguePage/ClassView.tsx

```diff
@@ -20,64 +24,77 @@ const filters = [
+  const onSearchInput = (event: any): void => {
+    setSearchBox(event.target.value);
+  };
 
-  return (
-    <>
-      {data.map((classs: Class) => (
-        <ClassCard
-          key={classs.eventInformationId}
-          name={classs.name}
-          minLevel={classs.minLevel}
-          maxLevel={classs.maxLevel}
-          rrstring={classs.rrstring}
-          startTime={classs.startTime}
-          endTime={classs.endTime}
-        />
-      ))}
-    </>
-  );
-};
+  const changeSort = (event: any): void => {
+    if (event.target.value === "alpha") {
+      setOrderBy({ alpha: true, level: false });
+    }
+    if (event.target.value === "level") {
+      setOrderBy({ alpha: false, level: true });
+    }
+  };
 
-const ClassView: React.FC = () => (
-  <div className={styles.compContainer}>
-    <div className={styles.leftContainer}>
-      <h1 className={styles.compTitle}>Classes</h1>
-      <div className={styles.compList}>
-        <ul className={styles.scroll}>
-          <ClassScroll />
-        </ul>
-      </div>
-    </div>
-    <span className={styles.spacer} />
-    <div className={styles.rightContainer}>
-      <input type="text" placeholder="Search classes" className={styles.searchBar}></input>
-      <h2 className={styles.orderTitle}>Order By:</h2>
-      <div className={styles.orderElem}>
-        <p className={styles.listItemText}>Alphabetical</p>
-        <input type="radio"></input>
+  const onFilterCheck = (event: any): void => {
+    const tempSelectedClassLevels = new Set(selectedClassLevels);
+    if (event.target.checked) {
+      tempSelectedClassLevels.add(Number(event.target.value));
+    } else if (!event.target.checked) {
+      tempSelectedClassLevels.delete(Number(event.target.value));
+    }
+    setSelectedClassLevels(tempSelectedClassLevels);
+  };
+
+  return (
+    <div className={styles.compContainer}>
+      <div className={styles.leftContainer}>
+        <h1 className={styles.compTitle}>Classes</h1>
+        <div className={styles.compList}>
+          <ul className={styles.scroll}>
+            <ClassScroll
+              searchQuery={searchBox}
+              orderBy={orderBy}
+              selectedLevels={selectedClassLevels}
+            />
+          </ul>
+        </div>
       </div>
-      <div className={styles.orderElem}>
-        <p className={styles.listItemText}>Level</p>
-        <input type="radio"></input>
+      <span className={styles.spacer} />
+      <div className={styles.rightContainer}>
+        <input
+          type="text"
+          placeholder="Search classes"
+          className={styles.searchBar}
+          onChange={onSearchInput}
+        ></input>
+        <h2 className={styles.orderTitle}>Order By:</h2>
+        <div className={styles.orderElem}>
+          <p className={styles.listItemText}>Alphabetical</p>
+          <input type="radio" value="alpha" checked={orderBy.alpha} onChange={changeSort}></input>
```

Comment:
instead of passing in the function, you can do something like this:
`onChange={ (e) => changeSort(e.target.value)`

Then, in the actual function, take in a `string` as one of the args instead of e. Its more type safe :)